packages/dynamic_layouts/lib/src/wrap_layout.dart

```diff
@@ -0,0 +1,271 @@
+/// do the same but it will fill the vertical axis and will pass to another
+/// column until it finishes.
+///
+/// This delegate creates grids with different sized tiles. But you
+/// can have equally sized tiles if you modify the `childCrossAxisExtent` and
+/// the `childMainAxisExtent` to be the same size.
+///
+/// See also:
+///  * [DynamicGridView.wrap], wich is a constructor to use this [SliverGridDelegate],
+///    like `GridView.extent`.
+///  * [DynamicGridView], which can use this delegate to control the layout of its
+///    tiles.
+///  * RenderDynamicSliverGrid], which can use this delegate to control the
+///    layout of its tiles.
+class SliverGridDelegateWithWrapping extends SliverGridDelegate {
+  /// Creates a delegate that makes dynamic grid layouts, the children Widgets
+  /// are the ones that decide their own size. Because the children decide their
+  /// own size the default values of `childMainAxisExtent` and
+  /// `childCrossAxisExtent` are set to [double.infinity].
+  /// If you only want to have the wrap, you can specify the `childCrossAxisExtent`
+  /// and the `childMainAxisExtent` to be the same size. Or only one of them to
+  /// be of a certain size in one of the axis.
+  const SliverGridDelegateWithWrapping({
+    this.mainAxisSpacing = 0.0,
+    this.crossAxisSpacing = 0.0,
+    this.childCrossAxisExtent = double.infinity,
+    this.childMainAxisExtent = double.infinity,
+    this.scrollDirection = Axis.vertical,
+  })  : assert(mainAxisSpacing != null && mainAxisSpacing >= 0),
+        assert(crossAxisSpacing != null && crossAxisSpacing >= 0);
+
+  /// The number of pixels from the leading edge of one tile to the trailing
+  /// edge of the same tile in the main axis.
+  ///
+  /// Defaults to [double.infinity].
+  final double childMainAxisExtent;
+
+  /// The number of pixels from the leading edge of one tile to the trailing
+  /// edge of the same tile in the cross axis.
+  ///
+  /// Defaults to [double.infinity].
+  final double childCrossAxisExtent;
+
+  /// The number of logical pixels between each child along the main axis.
+  ///
+  /// Defaults to 0.0
+  final double mainAxisSpacing;
+
+  /// The number of logical pixels between each child along the cross axis.
+  ///
+  /// Defaults to 0.0
+  final double crossAxisSpacing;
+
+  /// The axis along which the scroll view scrolls.
+  ///
+  /// Defaults to [Axis.vertical].
+  final Axis scrollDirection;
+
+  bool _debugAssertIsValid() {
+    assert(mainAxisSpacing >= 0.0);
+    assert(crossAxisSpacing >= 0.0);
+    return true;
+  }
+
+  @override
+  SliverGridLayout getLayout(SliverConstraints constraints) {
+    assert(_debugAssertIsValid());
+    return SliverGridWrappingTileLayout(
+      childMainAxisExtent: childMainAxisExtent,
+      childCrossAxisExtent: childCrossAxisExtent,
+      mainAxisSpacing: mainAxisSpacing,
+      crossAxisSpacing: crossAxisSpacing,
+      scrollDirection: scrollDirection,
```

Comment:
constraints.axisDirection :)

Reply:
axisDirectionToAxis(constraints.axisDirection) was the one to solve it but yeah that makes it way simpler XD